metaci/build/flows.py

```diff
@@ -9,24 +12,34 @@ class MetaCIFlowCallback(FlowCallback):
 
     def __init__(self, buildflow_id):
         self.buildflow_id = buildflow_id
-    
+
     def pre_task(self, step):
-        flowtask = FlowTask.objects.find_task(self.buildflow_id, step.path, step.step_num)
-        flowtask.description = step.task_config.get('description')
+        flowtask = FlowTask.objects.find_task(
+            self.buildflow_id, step.path, step.step_num
+        )
+        flowtask.description = step.task_config.get("description")
         flowtask.time_start = timezone.now()
-        flowtask.options = step.task_config['options']
+        flowtask.options = step.task_config["options"]
         flowtask.class_path = step.task_class.__name__
-        flowtask.status = 'running'
+        flowtask.status = "running"
+
         flowtask.save()
 
     def post_task(self, step, result):
-        flowtask = FlowTask.objects.find_task(self.buildflow_id, step.path, step.step_num)
+        flowtask = FlowTask.objects.find_task(
+            self.buildflow_id, step.path, step.step_num
+        )
         flowtask.time_end = timezone.now()
         flowtask.result = result.result
         flowtask.return_values = result.return_values
+
+        if step.task_class == Robot:
+            import_robot_test_results(flowtask, "output.xml")
```

Comment:
Part of the goal is to automatically find output.xml in the location where the robot task put it, and no longer require customizing the task in MetaCI to put it in the root folder of the repository.

Probably the cleanest way to do this is:
- Adjust the Robot task in cumulusci to set self.return_values["robot_outputdir"] to the outputdir path that was passed to robot itself.
- Then this code can do something like `if "robot_outputdir" in result.return_values:`

(This is a variation on what @prescod was suggesting, but indicating that the task has robot results in return_values instead of providing a method to access the results. I think the former will work better since we don't have the task instance anymore in this callback.)

Reply:
We now [set outputdir on the return_values](https://github.com/SFDO-Tooling/CumulusCI/pull/2252) for the Robot task.

Reply:
Seems like a great solution to me.